tests/dragonfly/replication_test.py

```diff
@@ -2096,3 +2086,42 @@ async def test_user_acl_replication(df_local_factory):
     await c_master.execute_command("ACL SETUSER tmp +replconf")
     await check_all_replicas_finished([c_replica], c_master, 5)
     assert 2 == await c_replica.execute_command("DBSIZE")
+
+
+@pytest.mark.asyncio
+async def test_replica_reconnect(df_local_factory):
+    # Connect replica to master
+    master = df_local_factory.create(proactor_threads=1)
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_master.execute_command("set k 12345")
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    assert not await is_replica_down(c_replica)
+
+    # kill existing master, create master with different repl_id but same port
+    master_port = master.port
+    master.stop()
+    assert await is_replica_down(c_replica)
+
+    master = df_local_factory.create(proactor_threads=1, port=master_port)
+    df_local_factory.start_all([master])
+
+    # Assert that replica did not reconnected to master with different repl_id
+    assert await c_master.execute_command("get k") == None
```

Comment:
so you dont need to call master.client() after stop() ?
cause I think we do this in all our test..

Reply:
nope, clients automatically reconnect